src/app/shared/picker/icon-picker/icons.ts

```diff
@@ -1253,11 +1377,1403 @@ export const brand: string[] = ['fab fa-500px',
+  ['tag', ['tag', 'label']],
+  ['tags', ['tags', 'labels']],
+  ['tape', ['tape']],
+  ['tasks', ['tasks', 'downloading', 'downloads', 'loading', 'progress', 'settings']],
+  ['taxi', ['taxi', 'cab', 'cabbie', 'car', 'car service', 'lyft', 'machine', 'transportation', 'uber', 'vehicle']],
+  ['teamspeak', ['teamspeak']],
+  ['telegram', ['telegram']],
+  ['telegram-plane', ['telegram-plane']],
+  ['tencent-weibo', ['tencent-weibo']],
+  ['tennis-ball', ['tennis-ball']],
+  ['terminal', ['terminal', 'code', 'command', 'console', 'prompt']],
+  ['text-height', ['text-height']],
+  ['text-width', ['text-width']],
+  ['th', ['th', 'blocks', 'boxes', 'grid', 'squares']],
+  ['th-large', ['th-large', 'blocks', 'boxes', 'grid', 'squares']],
+  ['th-list', ['th-list', 'checklist', 'completed', 'done', 'finished', 'ol', 'todo', 'ul']],
+  ['themeco', ['themeco']],
+  ['themeisle', ['themeisle']],
+  ['thermometer', ['thermometer', 'fever', 'temperature']],
+  ['thermometer-empty', ['thermometer-empty', 'status']],
+  ['thermometer-full', ['thermometer-full', 'status']],
+  ['thermometer-half', ['thermometer-half', 'status']],
+  ['thermometer-quarter', ['thermometer-quarter', 'status']],
+  ['thermometer-three-quarters', ['thermometer-three-quarters', 'status']],
+  ['thumbs-down', ['thumbs-down', 'disagree', 'disapprove', 'dislike', 'hand', 'thumbs-o-down']],
+  ['thumbs-up', ['thumbs-up', 'agree', 'approve', 'favorite', 'hand', 'like', 'ok', 'okay', 'success', 'thumbs-o-up', 'yes', 'you got it dude']],
+  ['thumbtack', ['thumbtack', 'coordinates', 'location', 'marker', 'pin', 'thumb-tack']],
+  ['ticket', ['ticket', 'movie', 'pass', 'support']],
+  ['ticket-alt', ['ticket-alt', 'ticket']],
+  ['times', ['times', 'close', 'cross', 'error', 'exit', 'incorrect', 'notice', 'notification', 'notify', 'problem', 'wrong', 'x']],
+  ['times-circle', ['times-circle', 'close', 'cross', 'exit', 'incorrect', 'notice', 'notification', 'notify', 'problem', 'wrong', 'x']],
+  ['times-hexagon', ['times-hexagon', 'close', 'cross', 'exit', 'incorrect', 'notice', 'notification', 'notify', 'problem', 'wrong', 'x']],
+  ['times-octagon', ['times-octagon', 'close', 'cross', 'exit', 'incorrect', 'notice', 'notification', 'notify', 'problem', 'wrong', 'x']],
+  ['times-square', ['times-square', 'close', 'cross', 'incorrect', 'notice', 'notification', 'notify', 'problem', 'window', 'wrong']],
+  ['tint', ['tint', 'drop', 'droplet', 'raindrop', 'waterdrop']],
+  ['tint-slash', ['tint-slash']],
+  ['tired', ['tired']],
+  ['toggle-off', ['toggle-off', 'switch']],
+  ['toggle-on', ['toggle-on', 'switch']],
+  ['toolbox', ['toolbox', 'admin', 'container', 'fix', 'repair', 'settings', 'tools']],
+  ['tooth', ['tooth']],
+  ['trade-federation', ['trade-federation']],
+  ['trademark', ['trademark']],
+  ['train', ['train', 'bullet', 'locomotive', 'railway']],
+  ['transgender', ['transgender', 'intersex']],
+  ['transgender-alt', ['transgender-alt']],
+
```

Comment:
This is very inefficient. The time complexity of the solution is `O(N*M)` where `N` is the number of icons and `M` is the maximum number of metadata items per icon. So basically, it executes around `7000` operations on every letter typed to search input.

It can be reduced to `O(1)` by reversing the map so meta words will become keys and the values will contain arrays of icons.

Reply:
It is not that easy. You want to search for a substring, so you still need to traverse the whole array of N icons...